R/FilteredDatasetDefault.R

```diff
@@ -1,269 +1,112 @@
-
-        varnames <- unique(unlist(lapply(state, "[[", "varname")))
-        private$get_filter_states()[[1]]$remove_filter_state(state)
-
-        logger::log_trace("{ class(self)[1] }$remove_filter_state removed filter(s), dataname: { private$dataname }")
-      })
-
+    #' @return `NULL`, invisibly
+    get_filter_overview = function() {
       invisible(NULL)
     },
 
+    ### shiny modules ----
+
     #' @description
-    #' UI module to add filter variable for this dataset
-    #'
-    #' UI module to add filter variable for this dataset
+    #' Overwrites parent method.
+    #' @details
+    #' Blank module UI that would list active filter states for this dataset.
     #' @param id (`character(1)`)\cr
-    #'  identifier of the element - preferably containing dataset name
-    #'
-    #' @return function - shiny UI module
-    ui_add = function(id) {
+    #'  `shiny` module id
+    #' @return empty `div`
+    ui_active = function(id) {
       ns <- NS(id)
-      tagList(
-        tags$label("Add", tags$code(self$get_dataname()), "filter"),
-        private$get_filter_states()[["filter"]]$ui_add(id = ns("filter"))
-      )
+      div()
     },
 
     #' @description
-    #' Get number of observations based on given keys
-    #' The output shows the comparison between `filtered_dataset`
-    #' function parameter and the dataset inside self
-    #' @return `list` containing character `#filtered/#not_filtered`
-    get_filter_overview = function() {
-      logger::log_trace("FilteredDataset$srv_filter_overview initialized")
-      # Gets filter overview subjects number and returns a list
-      # of the number of subjects of filtered/non-filtered datasets
-      subject_keys <- if (length(private$parent_name) > 0) {
-        private$join_keys
-      } else {
-        self$get_keys()
-      }
-
-      dataset <- self$get_dataset()
-      data_filtered <- self$get_dataset(TRUE)
-      if (length(subject_keys) == 0) {
-        data.frame(
-          dataname = private$dataname,
-          obs = nrow(dataset),
-          obs_filtered = nrow(data_filtered())
-        )
-      } else {
-        data.frame(
-          dataname = private$dataname,
-          obs = nrow(dataset),
-          obs_filtered = nrow(data_filtered()),
-          subjects = nrow(unique(dataset[subject_keys])),
-          subjects_filtered = nrow(unique(data_filtered()[subject_keys]))
-        )
-      }
+    #' Overwrites parent method.
+    #' @details
+    #' Blank module UI that would list active filter states for this dataset.
+    #' @param id (`character(1)`)\cr
+    #'  `shiny` module id
+    #' @return empty `div`
+    ui_add = function(id) {
+      ns <- NS(id)
+      div()
     }
   ),
   private = list(
-    parent_name = character(0),
-    join_keys = character(0)
+    ## Private Methods ----
+    add_filter_states = function(filter_states, id) {
+      warning("DefaultFilterState cannnot add filter states")
+    }
+
+    ## Private Fields ----
```

Comment:
Seems to be unnecessary. `add_filter_states` isn't called in this class nor in the parent (`FilteredDataset`) class. `private = list()` could be removed at all I guess

```suggestion
```

<img width="450" alt="Skärmavbild 2023-12-14 kl  13 25 32" src="https://github.com/insightsengineering/teal.slice/assets/6959016/404d52c4-8533-4f05-8385-b4cc74e83240">